### ISSUE TITLE
digitalocean_database_db - Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by wscourge** [digitalocean/terraform-provider-digitalocean#852]
# Bug Report

---

## Describe the bug

Error:

> Root resource was present, but now absent.

when creating db cluster in the root module and database in the child module

### Affected Resource(s)

- digitalocean_database_db
- digitalocean_database_cluster

### Expected Behavior

No error & succesful tf state update.

### Actual Behavior

> │ Error: Provider produced inconsistent result after apply
> │
> │ When applying changes to module.env.digitalocean_database_db.api, provider "provider[\"registry.terraform.io/digitalocean/digitalocean\"]" produced an
> │ unexpected new value: Root resource was present, but now absent.
> │
> │ This is a bug in the provider, which should be reported in the provider's own issue tracker.

The database is indeed created, and the terraform state does not update, resulting in the following error on the subsequent `terraform apply` run:

> │ Error: Error creating Database DB: POST https://api.digitalocean.com/v2/databases/61ce7960-f3a1-4ec6-a430-1d45c9cd60df/dbs: 422 (request "20e8baa1-af16-4af0-8ccd-4e2abf7763f0") database name is not available
> │
> │   with module.env.digitalocean_database_db.api,
> │   on modules/env/db.tf line 5, in resource "digitalocean_database_db" "api":
> │    5: resource "digitalocean_database_db" "api" {

as the name is no longer unique.

### Steps to Reproduce

1. Create `digitalocean_database_cluster` in the `root` module.
3. Create `digitalocean_database_db` in the `child` module.
4. Run `terraform apply`
5. See: error 1 (Root ...)
6. Run `terraform apply`
7. See error 2 (422)

**Terraform Configuration Files**

**Expected behavior**

**Debug Output**

**Panic Output**

## Additional context

**Important Factoids**

**References**

Probable cause is described in detail here: https://www.reddit.com/r/Terraform/comments/m5nv14/root_resource_was_present_but_now_absent/

TLDR: your `resourceDigitalOceanDatabaseDBCreate` function calls `return resourceDigitalOceanDatabaseDBRead(ctx, d, meta)` while it should not.

Thanks & keep up the good work.

**Comment by andrewsomething** [comment]
This seems to be the same issue as reported in this comment:

https://github.com/digitalocean/terraform-provider-digitalocean/issues/851#issuecomment-1199380378

We've raised this with the DBaaS team internally.

**Comment by dangerbat** [comment]
seems that the detail view of the API is affected. The list view it is working: 

```
GET /v2/databases/<clusterID>/dbs/ HTTP/2
Host: api.digitalocean.com
Accept: application/json
User-Agent: Terraform/1.1.8 godo/1.81.0
Authorization: Bearer <token>
Accept-Encoding: gzip, deflate
```

**Comment by andrewsomething** [comment]
This was indeed an issue with the detail view of the API. The DBaaS team has rolled out a fix, and the API should now be functioning as expected. I'm going to go ahead and close this out, but feel free to reopen if you're still running into problems.
